From b170e279b7575467aa7e4fa1e30a6af91d5e89ab Mon Sep 17 00:00:00 2001
From: Le Hoang Minh Thuan <thuanlhm@fsoft.com.vn>
Date: Thu, 22 Aug 2019 08:27:11 +0700
Subject: [PATCH 537/557] gpio-rcar: Prevent to force the interrupt handler as
 threaded

  On RT system, when kernel is booting with CONFIG_IRQ_FORCED_THREADING
 flag, the gpio_rcar_irq_handler() will be forced as threaded and then
 will generate a WARNING relating to interrupt handlers.
  To prevent to force the interrupt handler as threaded, we will add
 IRQF_NO_THREAD flag when requesting an interrupt for gpio-rcar driver.

Signed-off-by: Danh Duong <danhdc1@fsoft.com.vn>
Signed-off-by: Le Hoang Minh Thuan <thuanlhm@fsoft.com.vn>
---
 drivers/gpio/gpio-rcar.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpio/gpio-rcar.c b/drivers/gpio/gpio-rcar.c
index 5c65685..db9378f 100644
--- a/drivers/gpio/gpio-rcar.c
+++ b/drivers/gpio/gpio-rcar.c
@@ -570,7 +570,7 @@ static int gpio_rcar_probe(struct platform_device *pdev)
 
 	p->irq_parent = irq->start;
 	if (devm_request_irq(dev, irq->start, gpio_rcar_irq_handler,
-			     IRQF_SHARED, name, p)) {
+			     IRQF_SHARED | IRQF_NO_THREAD, name, p)) {
 		dev_err(dev, "failed to request IRQ\n");
 		ret = -ENOENT;
 		goto err1;
-- 
2.7.4

